Temporal Nexus: Landing page under Evaluate content (#3052)
temporalio/documentation: docs/evaluate/development-production-features/temporal-nexus.mdx

---
id: nexus
title: Temporal Nexus - Temporal feature
description: Explore how Temporal Nexus enables durable execution across team and namespace boundaries and promotes a modular architecture that enables each team to have their own namespace for improved security, troubleshooting, and blast radius isolation.
sidebar_label: Temporal Nexus
slug: /nexus
tags:
  - temporal-nexus
  - temporal-cloud

keywords:
  - temporal multi-team collaboration
  - modular temporal architecture
  - namespace boundary workflows
  - temporal nexus api registry
  - temporal nexus services
  - secure temporal workflows
  - multi-team integration temporal
  - temporal cloud nexus
---

> Temporal Nexus is a feature of the Temporal platform designed to connect durable executions across team, Namespace, region, and cloud boundaries - available in Public Preview for OSS and Temporal Cloud. 

Reviewer: ```suggestion
Temporal Nexus is a feature of the Temporal platform designed to connect durable executions across team, Namespace, region, and cloud boundaries - available in Public Preview for self-hosted deployments and Temporal Cloud. 
```

I would also add an asterisk on "cloud boundaries" and "self-hosted" to mention:
- Cross cloud calls are not available yet on Temporal Cloud
- Only single cluster deployments are supported for self-hosted

Temporal keeps the runtime benefits of an Event-Driven Architecture (EDA) where services can be down without affecting each other, but offers a higher-level durable execution programming model that lets you write idiomatic code in any language and preserve the state of execution.
This ensures that your Workflows can be resumed automatically in the presence of failure, without the extensive boilerplate code and complexity that comes with using raw EDA frameworks directly.

With Temporal Nexus, inter-service communication and durable execution can span teams, Namespaces, and regions all from within the Temporal context. 
Nexus preserves a familiar API programming model that supports both sync and async patterns, is suitable for low-latency and long-running operations, and uses an optimized form of the Temporal queue-based Worker architecture to preserve the benefits of EDA at runtime.

Temporal Nexus provides an integrated Temporal SDK experience, built-in Nexus Machinery, first-class observability, and enables each team to have their own Namespace for improved security, troubleshooting, and blast radius isolation.

> Temporal Nexus allows multiple teams to collaborate autonomously while maintaining a clean separation of concerns, ensuring that applications are scalable, secure, and easy to manage as they grow.

Author: should we remove this sentence? @prasek @irinabelova12

---
id: nexus
title: Temporal Nexus - Temporal feature
description: Explore how Temporal Nexus enables modular architecture and secure multi-team collaboration by abstracting Temporal primitives into clean Nexus Service APIs across namespace boundaries.
sidebar_label: Temporal Nexus
slug: /temporal-nexus
tags:
  - temporal-nexus
  - temporal-cloud

keywords:
  - temporal multi-team collaboration
  - modular temporal architecture
  - namespace boundary workflows
  - temporal nexus api registry
  - temporal nexus services
  - secure temporal workflows
  - multi-team integration temporal
  - temporal cloud nexus
---

> :::tip Support, stability, and dependency info

Reviewer 2: ```suggestion
```

> :::tip Support, stability, and dependency info

Author: Ended up adding it as apart of the first sentence instead of a banner since otherwise adding the banner will largen the scope.

> Temporal Nexus is available in [Public Preview](/evaluate/development-production-features/release-stages#public-preview) for both OSS and Temporal Cloud.

Reviewer 2: think we can omit this here as there are not many places in evaluate that mention OSS/open source since it's implicit as most features are in the open source, of note:
- major-components.mdx
- temporal-cloud/overview.mdx - mentions at the bottom of the overview page
- release-stages

self-host has more along with some in references or encyclopedia

```suggestion
```

Author: This also came out the convo with @irinabelova12. I do agree that it can be implied and otherwise, we'll have to make that differentiation for all of the pages which might not be neccessary at the moment.

:::

> Temporal Nexus is an integrated feature of the Temporal platform designed to connect durable executions across team and Namespace boundaries.

Reviewer 2: ```suggestion
Temporal Nexus is a new feature of the Temporal platform designed to connect durable executions across team, namespace, region, and cloud boundaries. It promotes a more modular architecture for sharing a subset of your team’s capabilities via well-defined service API contracts for other teams to use, that abstract underlying Temporal primitives, like Workflows, or execute arbitrary code.```

Author: taking away the word `new` since it will end up dating the docs.

> When organizations build applications, whether small or large, it’s essential to design them in a modular way to ensure scalability and adaptability. 

Reviewer 2: ```suggestion
Temporal keeps the runtime benefits of an Event-Driven Architecture (EDA) where services can be down without affecting each other, but offers a higher-level durable execution programming model that lets you write idiomatic code in any language and preserve the state of execution so it can be resumed automatically in the presence of failure, without the extensive boilerplate code and complexity that comes with using raw EDA frameworks directly.
```

> This involves breaking the application into smaller, manageable parts developed and operated by different teams. 

Reviewer 2: ```suggestion
With Temporal Nexus, inter-service communication and durable execution may now span teams, namespaces, and regions all from within the Temporal context. Nexus preserves a familiar API programming model that supports both sync and async patterns, is suitable for low-latency and long-running operations, and uses an optimized form of the Temporal queue-based worker architecture to preserve the benefits of EDA at runtime.
```

> While this modular approach supports growth and evolution without requiring major changes, existing solutions often result in tightly coupled Workflows, complex integrations, and security risks.

Reviewer 2: ```suggestion
Temporal Nexus provides an integrated Temporal SDK experience, built-in Nexus Machinery, first-class observability, and enables each team to have their own namespace for improved security, troubleshooting, and blast radius isolation.
```

> Teams collaborating on a Temporal Application often seek the isolation and security provided by separate Namespaces. However, they also need durable execution that works seamlessly across team and Namespace boundaries, ensuring secure integration with a clean separation of concerns.

Reviewer 2: ```suggestion
```

> Temporal Nexus allows multiple teams to collaborate autonomously while maintaining a clean separation of concerns, ensuring that applications are scalable, secure, and easy to manage as they grow.

Reviewer 2: collaborate autonomously?

```suggestion
```
